ui/redux/actions/publish.js

```diff
@@ -642,6 +676,25 @@ export const doPublish = (success: Function, fail: Function, preview: Function,
   }
 
   return Lbry.publish(publishPayload).then((response: PublishResponse) => {
+    // get the upload's claim id
+    const claimId = response.outputs.find((obj) => {
+      // $FlowFixMe
+      return obj.claim_id;
+      // $FlowFixMe
+    }).claim_id;
+
+    // hit backend to save restricted memberships
+    if (restrictedToMemberships || restrictedToMemberships === '') {
+      dispatch(doSaveMembershipRestrictionsForContent(channelClaimId, claimId, restrictedToMemberships));
+    }
+
+    dispatch(
+      doUpdatePublishForm({
+        restrictedToMemberships: undefined,
+        restrictCommentsAndChat: undefined,
+      })
+    );
```

Comment:
:question: In the case of `preview` (to get estimated LBC cost), seems like this will be hit.  Is the backend being hit twice?

:small_blue_diamond: Would it be possible to move this into `publishSuccess`?  `doPublish` is meant to be a thin wrapper to generate the payload and make the call. Anything else should be at the callback level.

Reply:
Looks like this is already "publish success" and preview is handled in the block above it?

Reply:
To clarify, this is just pre-refactoring for the future;  current version is fine if there's no time.